Compiling on linux arm64

Hey, title says it all so I'll keep it short, I'm compiling on an m2 macbook running fedora linux asahi remix and this is the error I get:

```
cc1plus: error: unknown value ‘nehalem’ for ‘-march’
cc1plus: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a armv8.6-a armv8.7-a armv8.8-a armv8.9-a armv8-r armv9-a armv9.1-a armv9.2-a armv9.3-a armv9.4-a armv9.5-a native
gmake[3]: *** [src/scxt-core/CMakeFiles/scxt-core.dir/build.make:79: src/scxt-core/CMakeFiles/scxt-core.dir/browser/browser.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:2250: src/scxt-core/CMakeFiles/scxt-core.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:1276: CMakeFiles/shortcircuit-products.dir/rule] Error 2
gmake: *** [Makefile:170: shortcircuit-products] Error 2
```
shitty workarounds, or if it is worth fixing, a commit to master would be much appreciated.

--- a/cmake/compiler-options.cmake
+++ b/cmake/compiler-options.cmake
@@ -74,7 +74,7 @@ elseif (UNIX AND NOT APPLE)
             -Werror
             -Wno-multichar
             -fPIC  # probably not needed with CMAKE_POSITION_INDEPENDENT_CODE_ON but hey
-            -march=nehalem
+            $<IF:$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},aarch64>,-march=armv8-a,-march=nehalem>
             )
     set(OS_COMPILE_DEFINITIONS
             LINUX=1)

